src/ray/core_worker/core_worker.h

```diff
@@ -557,6 +556,8 @@ class CoreWorker {
   // Interface to submit non-actor tasks directly to leased workers.
   std::unique_ptr<CoreWorkerDirectTaskSubmitter> direct_task_submitter_;
 
+  mutable std::mutex actor_handles_mutex_;
```

Comment:
If you're going to add a lock, please use absl::Mutex and use thread annotations to show which fields are guarded by the mutex.

Reply:
Sure, I can do that.

Reply:
Done.